libs/sdk-ui-ext/src/internal/components/configurationControls/comparison/calculation/CalculationListItem.tsx

```diff
@@ -0,0 +1,52 @@
+// (C) 2023 GoodData Corporation
+import React, { useMemo } from "react";
+import cx from "classnames";
+
+import { ISingleSelectListItemProps, BubbleHoverTrigger, Bubble } from "@gooddata/sdk-ui-kit";
+
+import CalculationListItemInfo from "./CalculationListItemInfo.js";
+import { HIDE_DELAY_DEFAULT, SHOW_DELAY_DEFAULT } from "../../../../constants/bubble.js";
+
+const BUBBLE_INFO_ALIGN_POINTS = [{ align: "cr cl" }];
+const BUBBLE_INFO_ARROW_OFFSETS = { "cr cl": [15, 0] };
+
+const CalculationListItem: React.FC<ISingleSelectListItemProps> = ({
+    title,
+    icon,
+    info,
+    isSelected,
+    onClick,
+}) => {
+    const classNames = useMemo(
```

Comment:
This probably does not need to be memoized as it produces string. It's similar to line 30 where you have template string but do not use memeoization.

Reply:
Removed useMemo!

By the way, I'm still a bit perplexed about how to effectively use useCallback and useMemo. Despite spending a significant amount of time contemplating this matter, my understanding hasn't improved considerably. Therefore, I'd like to inquire if there are any best practices for making the most of useMemo and useCallback.  Acquiring such insights would not only enhance my understanding but also enable me to perform better in the future.

Thank you so much!

Reply:
I struggle with this sometimes as well :) There are people in GD who  obsessively memoize everything. Supposedly this is bad as the memoization costs some overhead as well. If it would be necessary, framework would do memoization under the hood for everything. The rule is that you should not memoize anything, unless it is obviously expensive (lots of object transformations, API calls, etc.), measure the performance, analyze number of re-renders and then optimize as needed. This is hard and we don't do it. Therefore it is up to everyone to decide if what you do is expensive and deserve memoization or what component below in the tree does is expensive and must not rerender. Also, the change of object signature does not mean component will re-render. It's up to React to decide if re-render is necessary even when prop signature changed.

Reply:
Thank you for your knowledge-sharing, this help me a lot!